<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DH_UI</class>
 <widget class="QWidget" name="DH_UI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>364</width>
    <height>264</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QPushButton" name="back_button">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>返回主菜单</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="generate_PG_layout">
     <item>
      <widget class="QPushButton" name="generate_PG_button">
       <property name="baseSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>生成&lt;p, g&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="generate_PG_lineEdit">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="generate_Akey_layout">
     <item>
      <widget class="QPushButton" name="generate_Akey_button">
       <property name="text">
        <string>A生成一个随机数a</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="generate_Akey_lineEdit">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="generate_Bkey_layout">
     <item>
      <widget class="QPushButton" name="generate_Bkey_button">
       <property name="text">
        <string>B生成一个随机数b</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="generate_Bkey_lineEdit">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="send_ga_layout">
     <item>
      <widget class="QPushButton" name="send_ga_button">
       <property name="text">
        <string>A发送g^a % p给B</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="send_ga_lineEdit">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="send_gb_layout">
     <item>
      <widget class="QPushButton" name="send_gb_button">
       <property name="text">
        <string>B发送g^b % p给A</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="send_gb_lineEdit">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="cal_A_share_key_layout">
     <item>
      <widget class="QPushButton" name="cal_A_share_key_button">
       <property name="text">
        <string>A计算共享密钥</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="cal_A_share_key_lineEdit">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="cal_B_share_key_layout">
     <item>
      <widget class="QPushButton" name="cal_B_share_key_button">
       <property name="text">
        <string>B计算共享密钥</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="cal_B_share_key_lineEdit">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
